SubmissionTest.java

```diff
@@ -148,15 +148,21 @@ public void testZip() throws IOException
         
         // Get actual Submission filenames.
         File actualDir = new File("zips");
-        File[] actualZips = actualDir.listFiles();
-        Arrays.sort(actualZips);
-        String[] actual = new String[actualZips.length];
-        for (int i = 0; i < actual.length; i++)
+        
+        // Make sure directory has not been deleted 
+        // by deleteZips test or application. 
+        if (actualDir.exists())
         {
-            actual[i] = actualZips[i].getName();
-        }
+            File[] actualZips = actualDir.listFiles();
+            Arrays.sort(actualZips);
+            String[] actual = new String[actualZips.length];
+            for (int i = 0; i < actual.length; i++)
+            {
+                actual[i] = actualZips[i].getName();
+            }
 
-        assertArrayEquals("should be same", expected, actual);
+            assertArrayEquals("should be same", expected, actual);
```

Comment:
@haietza I think for now I feel most comfortable with leaving it how it is, with the occasional failure of the two tests.  Making these independent without detracting from their usefulness might require more heavy redesign (or be impossible, you know better than I).  Is that okay with you?